File: go/secret-handshake/secret_handshake.go
package secret

const (
	wink = 1 << iota
	doubleBlink
	closeEyes
	jump
	reverse
)

func Handshake(code int) []string {
	commands := []string{}

	if code < 0 || code >= reverse*2 {
		return commands
	}

	if code&wink != 0 {
		commands = append(commands, "wink")
	}

	if code&doubleBlink != 0 {
		commands = append(commands, "double blink")
	}

	if code&closeEyes != 0 {
		commands = append(commands, "close your eyes")
	}

	if code&jump != 0 {
		commands = append(commands, "jump")
	}

	if code&reverse != 0 {
		for i, j := 0, len(commands)-1; i < j; i, j = i+1, j-1 {
			commands[i], commands[j] = commands[j], commands[i]
		}
	}

	return commands
}

File: fsharp/secret-handshake/SecretHandshake.fs
module SecretHandshake

type Sign =
    | Wink = 0b1
    | DoubleBlink = 0b10
    | CloseEyes = 0b100
    | Jump = 0b1000
    | Revert = 0b10000

let handshake number =
    let translate list sign =
        match sign with
        | Sign.Wink -> "wink" :: list
        | Sign.DoubleBlink -> "double blink" :: list
        | Sign.CloseEyes -> "close your eyes" :: list
        | Sign.Jump -> "jump" :: list
        | _ -> List.rev list

    let powersOfTwo = [ for i in [0 .. 5] -> 1 <<< i]

    [ for p in powersOfTwo do if p &&& number > 0 then yield enum<Sign> p ]
        |> List.fold translate []
        |> List.rev